src/permission/fs_permission.cc

```diff
@@ -0,0 +1,209 @@
+#include "fs_permission.h"
+#include "base_object-inl.h"
+#include "v8.h"
+
+#include <fcntl.h>
+#include <limits.h>
+#include <stdlib.h>
+#include <algorithm>
+#include <filesystem>
+#include <iostream>
+#include <string>
+#include <vector>
+
+namespace {
+
+std::string WildcardIfDir(const std::string& res) noexcept {
```

Comment:
Why is this necessary?

Reply:
The FSPermission uses a PrefixRadixTree to handle allowed resources. When you pass a directory, we store it as `dir/*` to use the wildcard child so anything after `dir/` will be allowed (e.g: `dir/foo` or `dir/bar`).